lib/Rest/HttpRequest.cpp

```diff
@@ -545,9 +545,24 @@ void HttpRequest::setHeaderV2(std::string key, std::string value) {
   
   if (key == StaticStrings::Accept && value == StaticStrings::MimeTypeVPack) {
     _contentTypeResponse = ContentType::VPACK;
-  } else if (key == StaticStrings::ContentTypeHeader && value == StaticStrings::MimeTypeVPack) {
-    _contentType = ContentType::VPACK; // don't insert this header!!
-    return;
+  } else if ((_contentType == ContentType::UNSET) && (key == StaticStrings::ContentTypeHeader)) {
+    if (value == StaticStrings::MimeTypeVPack) {
+      _contentType = ContentType::VPACK; // don't insert this header!!
+      return;
+    }
+    else if (value == StaticStrings::MimeTypeJsonNoEncoding) {
+      _contentType = ContentType::JSON;
+      return;
+    }
+    else if ((value.length() >= StaticStrings::MimeTypeJsonNoEncoding.length()) &&
+             (memcmp(value.c_str(),
+                     StaticStrings::MimeTypeJsonNoEncoding.c_str(),
+                     StaticStrings::MimeTypeJsonNoEncoding.length()) == 0)) {
+      // ignore encoding etc.
+      _contentType = ContentType::JSON;
+      return;
+    }
+    _contentType = ContentType::UNSET;
```

Comment:
this line is superfluous, as we can only get here if `_contentType` is `ContentType::UNSET` already.